server.go

```diff
@@ -458,8 +461,18 @@ func (srv *Server) serveUDP(l *net.UDPConn) error {
 
 	rtimeout := srv.getReadTimeout()
 	// deadline is not used here
+	last := time.Now()
+	pkts := uint64(0)
+	opt := &AcceptOption{}
 	for srv.isStarted() {
 		m, s, err := reader.ReadUDP(l, rtimeout)
+		pkts++
+		if srv.PacketAccepter != nil && pkts%100 == 0 {
```

Comment:
This should all go after both the `err != nil` and `len(m) < headerSize` checks. The error one because we might not even have a packet at this point, and the `len(m)` because we just assume they're junk and ignore them unconditionally.